resources/building-mobile-extensions/building-an-extension.md

```diff
@@ -23,15 +23,69 @@ Before you build an extension, complete the following tasks:
 
   Extensions for iOS can currently only be built using Objective-C, and extensions on Android can currently only be built using Java.
 
-* Ensure that you have already included the Adobe Cloud Platform SDKs in your project.
+* Ensure that you have already included the Adobe Experience Platform SDKs in your project.
 
-## Implementing an Adobe Cloud Platform SDK Extension
+## Implementing an Adobe Experience Platform SDK Extension
 
 To create a simple extension, complete the following procedures in the order in which they are listed:
 
 ### A. **Create an Extension Class**
 
-The `ACPExtension`\(iOS\) or `Extension` \(Android\) class is the base class that any extensions must derive from. The `init` method \(iOS\) or the base `constructor` \(Android\) of your extension class is where you will have the opportunity to extend the Adobe Cloud Platform SDKs functionality by registering event listeners, or by setting a default shared state that other modules can access.
+The `ACPExtension`\(iOS\) or `Extension` \(Android\) class is the base class that any extensions must derive from. The `init` method \(iOS\) or the base `constructor` \(Android\) of your extension class is where you will have the opportunity to extend the Adobe Experience Platform SDKs functionality by registering event listeners, or by setting a default shared state that other modules can access.
+
+{% tabs %}
+{% tab title="Android" %}
+
+#### **Android**
+
+The `Extension` class has the following method that you must override:
+
+- `getName`, which returns the name of the extension.
+
+  Extension developers must prefix their extension names with the company name \(for example, _com.myCompany.myExtension_\). For more information about the naming constraints, see Namespace Conventions. The name that you use to register cannot conflict with other registered extensions or Adobe internal modules.
+
+**Tip**: All Adobe module names are prefixed with _com.adobe.module_ and are considered reserved.
+
+- `onUnregistered`, which allows your extension to complete the cleanup that is required when the Adobe Experience Platform SDK unregisters your extension.  Unregistration typically happens at app shutdown but can also occur when an extension is behaving badly. Examples of the extension behaving badly include taking too long to handle a callback or by throwing an exception.
```

Comment:
`onUnregistered` is not required to be overridden. Move this down to the next section of "optionally override" methods.